apps/browser/qml/pages/components/TabView.qml

```diff
@@ -80,32 +75,34 @@ Item {
             TabGridView {
                 id: _tabView
                 privateMode: false
-                portrait: root.portrait
+                portrait: tabView.portrait
                 model: tabItem.privateMode ? webView.privateTabModel : webView.persistentTabModel
                 header: Item {
                     width: 1
-                    height: tabs.tabBarHeight + Theme.paddingLarge
+                    height: Theme.paddingLarge
                 }
 
-                onHide: root.hide()
-                onEnterNewTabUrl: root.enterNewTabUrl()
-                onActivateTab: root.activateTab(index)
-                onCloseTab: root.closeTab(index)
-                onCloseAll: root.closeAll()
-                onCloseAllCanceled: root.closeAllCanceled()
-                onCloseAllPending: root.closeAllPending()
+                onHide: tabView.hide()
+                onEnterNewTabUrl: tabView.enterNewTabUrl()
+                onActivateTab: tabView.activateTab(index)
+                onCloseTab: tabView.closeTab(index)
+                onCloseAll: tabView.closeAll()
+                onCloseAllCanceled: tabView.closeAllCanceled()
+                onCloseAllPending: tabView.closeAllPending()
             }
+
             onIsCurrentItemChanged: {
                 if (isCurrentItem) {
                     _remorsePopup = Qt.binding(function() { return _tabView.remorsePopup })
                 }
             }
 
-            Rectangle {
-                color: Theme.colorScheme == Theme.LightOnDark ? "black" : "white"
-                width: parent.width
-                height: Math.max(0, _pulleyMenuHeight - y)
-                y: Math.max(0, _pulleyMenuHeight - _tabView.contentY + _tabView.originY)
+            Connections {
+                target: popupMenu
+                onCloseAllTabs: {
+                    if (isCurrentItem)
+                        _tabView.closeAllTabs()
+                }
```

Comment:
I notice that the `popupMenu.closeAllTabs` signal goes into the `tabGridView`, then comes out again and gets signalled all the way out to `Overlay`. I've not tried this, so just a suggestion, but it might be worth checking whether this can be simplified by moving the `TabGridView.closeAllTabs()` method out of `TabGridView` and into `TabView`. From what I can tell, it's only needed inside `TabGridView` in order to set the `closingAllTabs` property there.

Reply:
I tried to fix it, it stops working with this approach

Reply:
I was hoping it might be straightforward, but can appreciate I probably missed some underlying complexity. Thank you for trying it out.